app/views/collection_numbers/_form.html.erb

```diff
@@ -1,29 +1,29 @@
-<% # form_tag(add_query_param(action: :new, id: @observation.id)) do %>
-<%= form_with model: :collection_number do |f| %>
+<% url =
+     "#{collection_numbers_path}/#{@observation.id}?q=#{get_query_param}" %>
+<%= form_with(model: @collection_number,
```

Comment:
Is there a more idiomatic way to do this?

Reply:
Hi Joe- sorry late to the thread! Yes there is. Will supply via email

Reply:
You're not late at all. I just started the thread today. It will take me a while to get this ready for review. (Of course, any interim comments are welcome, but you probably have better things to do.)

Reply:
This shouldn't need a url with the new form_with model syntax, but it will need the @observation.id as a param, i'll post a reference for that syntax in a minute. 

Rails 5.2 just knows that if it's a form_with model: @collection_number, the url is the new_collection_number_path or edit_collection_number_path.

Reply:
I've actually got the form rewritten with new syntax on my current branch. Here it is. 
The observation.id, necessary for a new collection number, is passed as a hidden field- that's a new thing.

```
<%= form_with model: :collection_number do |f| %>

  <%= f.submit button.t, class: "btn mx-auto mt-3" %>

  <%= f.hidden_field :id, value: @observation.id %>
  <%= f.hidden_field :q, value: get_query_param %>

  <% if @collection_number.observations.size > 1 %>
    <div class="alert alert-warning mt-3">
      <%= :edit_affects_multiple_observations.t(type: :collection_number) %>
    </div>
  <% end %>

  <div class="form-group mt-3">
    <%= f.label :name, :collection_number_name.t + ":") %>
    <span class="help-note">(<%= :required.t %>)</span>
    <%= f.text_field :name, class: "form-control" %>
  </div>

  <div class="form-group mt-3">
    <%= f.label :number, :collection_number_number.t + ":" %>
    <span class="help-note">(<%= :required.t %>)</span>
    <%= f.text_field :number, class:"form-control" %>
  </div>

  <%= f.submit button.t, class: "btn mx-auto mt-3" %>

<% end %>
```

Reply:
You'll see the new syntax everywhere in this one. All the labels and fields are bound to the |f| instance, and most parentheses are gone.

Reply:
I'll put a bunch of new controller/view changes up in a few minutes and it may be helpful to merge those into what you're working on. I haven't touched any tests except locations/descriptions, but there is a flotilla of changes throughout views, and some in controllers and helpers. I've gone through them file by file, analphabetically (ok, ok, that's my idea of a good malapropism) and i'm almost back to the top of the alphabet. Will push as soon as that's done.

Reply:
FYI: There's a stray close parenthesis on this line:
> ```
>     <%= f.label :name, :collection_number_name.t + ":") %>
> ```